### DISCUSSION TITLE
expanded html guidance details

### DISCUSSION THREAD

**Discussion by briesenberg07** [uwlib-cams/sinopia_maps#119] [technical]
Details on providing expanded guidance text, links, etc. in [the HTML versions of templates](https://uwlib-cams.github.io/sinopia_maps/) will go here as they are available.

REFERENCE:
- Related issues #113 + #123 
- Information about the prop_set > guidance_set and child elements and guidance data-entry in prop_set - see [DRAFT_guidance_set](https://docs.google.com/document/d/1s7wBUa525j4h8RYV0BON1UBPVHW3NiZJpxOUF3Jx4Fc/edit?usp=sharing)
- The RDA REgistry provides a [Map from RDA properties to recording methods](http://www.rdaregistry.info/Maps/mapRDA2Path.html).

**Comment by briesenberg07** [comment]
# 2023-04-30
**&#x1f6a7; WORK IN PROGRESS &#x1f6a7;**  

@pan-zhuo 
CC @cspayne (this shouldn't affect Python workflow much, we will just run a different stylesheet in 001_output_publish_load.py)

## ~~Proposed~~ Updated proposed method for outputting guidance
- Add `<uwsinopia:guidance_set>` as a child element of `<uwmaps:sinopia>`
- guidance_set has several child elements
- Attribute `@rt_id` required for each, with either value 'default' or RT ID for custom guidance; stylesheet matches on default if no custom guidance
- See `uwsinopia.xsd` > guidance_set_type

## Processing mixed content
### 2023-04-30: Unable to output HTML5 tags in the output using xsl:copy-of
- `<p/>`, `<strong/>`, and `<a/>` elements are output as `<uwsinopia:p/>`, `<uwsinopia:strong/>`, and `<uwsinopia:a/>`, and do not function as desired for browser display
- Everything I've looked at regarding importing elements from the HTML5 namespace into an .xsd file looks involved. 
- Methods I've tried with xsl:copy-ofwhich do *not* involve importing the HTML5 namespace into my schema:
   - `<xsl:copy-of copy-namespaces="no">`
   - `<xsl:copy-of copy-namespaces="no" validation="strip">`\*
   - `<xsl:stylesheet [..] exclude-result-prefixes="uwsinopia">`

\* Tried this despite having very-little understanding of this attribute's function 😓 

### 2023-05-02: Processing mixed content 
Successful using [xsl:apply-templates](https://github.com/uwlib-cams/sinopia_maps/blob/e7f22fa448ab3b40a56095039a79a722f3b8ab10/xsl/004a_01_prop_set_to_html.xsl#L229-L256) and [simple templates for guidance set child elements](https://github.com/uwlib-cams/sinopia_maps/blob/e7f22fa448ab3b40a56095039a79a722f3b8ab10/xsl/004a_01_prop_set_to_html.xsl#L286-L316).

**Reply by briesenberg07** [reply]
I should perhaps add that the new stylesheet and schema updates are being made in branch [expand-fix-html-rts](https://github.com/uwlib-cams/sinopia_maps/tree/expand-fix-html-rts)

**Reply by briesenberg07** [reply]
# Follow up on effects on Python workflow 
- @cspayne I was wrong, I believe these changes *will* affect Python code; for example map_storage/py/[update_prop_sets.py](https://github.com/uwlib-cams/map_storage/blob/main/py/update_prop_sets.py) will need to process guidance sets in addition to implementation sets when updating prop_sets

**Reply by briesenberg07** [reply]
Hello @pan-zhuo 
Added additional formatting for guidance text, hopefully this will meet all the needs. I wanted to allow for emphasizing text and for adding links. Also allows for numbered and unnumbered lists which may be nested. See [test template > has book artist person > guidance](https://uwlib-cams.github.io/sinopia_maps/html/UWSINOPIA_WAU_rdaWork_test_ries07.html#uwmaps_rdaw_p10k_wP10428_is_d2e855) for example formatting.

**Reply by briesenberg07** [reply]
See also [xsd for guidance_set element](https://github.com/uwlib-cams/sinopia_maps/blob/6ac61c6a426b7c75ef35f445e3a8de83622a82cc/xsd/uwsinopia.xsd#L29-L74)

**Comment by pan-zhuo** [comment]
Thank you Benjamin. I like the idea of separating the guidance set from the implementation set. I'm wondering if there are prompts/validation for @rt_id values?

**Reply by briesenberg07** [reply]
Thanks @pan-zhuo  
There will be validation for \@rt_id. Options will include 'default', and RT_IDs pulled from sinopia_maps.xml. (This is different from the example I showed above, where the 'default' guidance_set had no \@rt_id.)

**Comment by briesenberg07** [comment]
Updated HTML templates are live!  
Additional details available in https://github.com/uwlib-cams/sinopia_maps/issues/113
